### PR TITLE
Remove the unused stage implementation from Firebase Performance.

### PR DESCRIPTION
**Pull request by visumickey**

### CODE REVIEW

File: FirebasePerformance/Tests/TestApp/Podfile [hunk 1]
```diff
@@ -5,6 +5,8 @@ platform :ios, '11.0'
 #source 'sso://cpdc-eap/spec'
 #source 'https://github.com/CocoaPods/Specs.git'
 
+use_modular_headers!
```
Comment (jeremyjiang-dev) [inline FirebasePerformance/Tests/TestApp/Podfile:8]: Curious on what's the purpose of this?

**Review by jeremyjiang-dev — APPROVED** [review]
Thanks for cleaning up stages on iOS!

File: FirebasePerformance/Tests/TestApp/Podfile (hunk 1 above)
Comment (visumickey) [reply FirebasePerformance/Tests/TestApp/Podfile:8]: tl;dr is that - Cocoapods way of organizing dependencies (SDKs) are different from how Swift looks at them as modules. This construct, enables a seamless compatibility of Cocoapods based dependencies with the Swift modules.

A nice read about "Modular headers" and the need for this construct is at https://blog.cocoapods.org/CocoaPods-1.5.0/.

Comment (paulb777) [reply FirebasePerformance/Tests/TestApp/Podfile:8]: `use_modular_headers!` is now somewhat deprecated by `use_frameworks! :linkage => :static`. See https://blog.cocoapods.org/CocoaPods-1.9.0-beta/

File: FirebasePerformance/Tests/Unit/Timer/FIRTraceTest.m [hunk 2]
```diff
@@ -126,68 +126,6 @@ - (void)testTraceCreationWhenSDKFlagEnabledWithDataCollectionDisabled {
...
-  OCMStub([mock logTrace:[OCMArg any]]).andDo(nil);
-  [trace start];
-  [trace startStageNamed:@"1"];
-  [[mock reject] logTrace:trace.activeStage];
-  [trace startStageNamed:@"2"];
-  [trace stop];
-}
-
-/** Validates that stopping a trace does trigger an event being sent to Fll. */
-- (void)testTraceStopDoesTriggerEventSend {
```
Comment (raymondlam) [inline FirebasePerformance/Tests/Unit/Timer/FIRTraceTest.m:180]: Do we test trace stop elsewhere?

**Review by raymondlam — APPROVED** [review]

**Review by leotianlizhan — APPROVED** [review]

**Comment by paulb777** [comment]
Please close, merge, or comment. We plan to close stale PRs on November 28, 2023.

**Pull request closed without merging**